### DISCUSSION TITLE
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file

### DISCUSSION THREAD

**Discussion by sayak340** [apache/airflow#27692] [Q&A]
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file
when i run airflow db init its continue showing this error also unitests.cfg file also not created after i try to create database.

**Comment by potiuk** [comment]
There is no way to help you with that - you have problem with your sqlite db. Likely permission issue - to both create db and config. You need to  fixi it. But How - only you can know. Just make sure whatever you run airflow with have read/write permission to the HOME and AIRFLOW_HOME directories.

**Reply by sayak340** [reply]
this is my first-time using airflow ,i never use it before so i am confused now if i doing something wrong from beginning, can you guide me or tell me how i should have install airflow so i can fix this error, or can give me some reference.

**Reply by sayak340** [reply]
the above issue change to this WARNI [airflow.models.crypto] empty cryptography key - values will not be stored encrypted.
Initialization done

**Reply by codecakes** [reply]
@sayak340 how did you resolve the first issue?

**Comment by sayak340** [comment]
Hello sir
        The issue regarding sql light has been resolved. Thank you for your
kind reply. Currently I am facing a new issue (WARNI
[airflow.models.crypto] empty cryptography key - values will not be stored
encrypted. Initialization done) help me if possible.
Thank you
Sayak

**Reply by potiuk** [reply]
This is a warning. If you need encrypted values you need to fix it - look at our docs and search for "cryptography". You will find answers there. Our documentation is pretty comprehensive and well searchable and it's aimed to answer a lot of the problems you might have - as long as you use "search" there you will be able to find your answers there rather easily.

**Reply by potiuk** [reply]
And general comment for the future - Airflow documentation is meant to help with  "self-learning". It does not scale to teach every new users the basics individually and asnwer such needs when you stumble on somethign already explained an searchable. If you need to learn more about Airflow and how it works in a "tutorial" way - there are quite a number of great tutorials and courses available for step-by-step learning. 

A number of people invested in docs and videos and tutorials in order to guide you in your self-learning journey. See https://airflow.apache.org/ecosystem/#learning-resources

I suggest to use those resources rather than using discussions to learn the basics - it does not really scale.

**Comment by marchenko1985** [comment]
Hello, I got the same error, the reason was my wish to not mess up my home directory, so I did:

```bash
export AIRFLOW_HOME=./airflow
```

instead of

```bash
export AIRFLOW_HOME=~/airflow
```

and exactly that caused this error on next steps


Did tried to add small note into docs

https://github.com/apache/airflow/pull/36768


Hopefully will be merged, will help a lot in future, because it is not obvious

**Reply by jscheffl** [reply]
#36774 is the counter proposal.

**Reply by potiuk** [reply]
Or rather spin-off :)
